src/sparseml/pytorch/optim/modifier_pruning.py

```diff
@@ -607,19 +618,70 @@ def mask_type(self, value: Union[str, List[int], PruningMaskCreator]):
         if not isinstance(value, PruningMaskCreator):
             self._mask_creator = load_mask_creator(value)
 
+    @ModifierProp()
     def global_sparsity(self) -> bool:
         """
         :return: True if global pruning is enabled, False otherwise
         """
         return self._global_sparsity
 
+    @ModifierProp()
+    def score_type(self) -> PruningScoreTypes:
+        """
+        :return: the the scoring method used for pruning
+        """
+        return self._score_type
+
     @ModifierProp(serializable=False)
     def applied_sparsity(self) -> float:
         """
         :return: the currently applied sparsity level to the contained params
         """
         return self._applied_sparsity
 
+    def optimizer_pre_step(
+        self, module: Module, optimizer: Optimizer, epoch: float, steps_per_epoch: int
+    ):
+        """
+        Update mask movement scores with gradients right before optimizer step is
+        applied. Called here in case gradients are changed between the backwards
+        pass and step such as in grad norm clipping
+        :param module: module to modify
+        :param optimizer: optimizer to modify
+        :param epoch: current epoch and progress within the current epoch
+        :param steps_per_epoch: number of steps taken within each epoch
+            (calculate batch number using this and epoch)
+        """
+        super().optimizer_pre_step(module, optimizer, epoch, steps_per_epoch)
+
+        if self._module_masks.score_type == PruningScoreTypes.MOVEMENT:
```

Comment:
Given this is specific for movement pruning, the MovementPruningModifier should override this optimizer_pre_step function.

Reply:
Let's push comments like these off to a refactor later. Priority is to get this landed ASAP for release.

Reply:
this was refactored in the woodfisher PR to call into a single callback from the pruning mask object